cachito/workers/pkg_managers/npm.py

```diff
@@ -0,0 +1,120 @@
+# SPDX-License-Identifier: GPL-3.0-or-later
+import json
+import logging
+import os
+
+from cachito.errors import CachitoError
+from cachito.workers.pkg_managers.general_js import download_dependencies
+
+__all__ = ["get_package_and_deps", "resolve_npm"]
+
+log = logging.getLogger(__name__)
+
+
+def _get_deps(package_lock_deps, name_to_deps):
+    """
+    Get a mapping of dependencies to all versions of the dependency.
+
+    This function works by populating the input ``name_to_deps``. When first called, this value
+    should be ``{}``. As the function executes, it will call itself by passing in the current value
+    of ``name_to_deps`` for instances where the dependency tree is more than a level deep. This
+    occurs when dependency A and B depend on different versions of the dependency C.
+
+    ``name_to_deps`` is a dictionary so that deduplication is much more efficient than if it were
+    a list. If it were a list, there'd be a time complexity of O(n) every time a dependency is to be
+    inserted.
+
+    :param dict package_lock_deps: the value of a "dependencies" key in a package-lock.json file
+    :param dict name_to_deps: the current mapping of dependencies
+    :raise CachitoError: if the lock file contains a dependency not from the registry
+    """
+    for name, info in package_lock_deps.items():
+        # Note that a bundled dependencies will not have the "resolved" key, but those are supported
+        # since they are properly cached in the parent dependency in Nexus
```

Comment:
If a dependency has bundled dependencies, they will not have the "resolved" key set. This is okay since they are included in the dependency that bundled them.

I'll clarify the comment.